### DISCUSSION TITLE
Comments feature for PDF

### DISCUSSION THREAD

**Discussion by Rots** [xournalpp/xournalpp#2681] [Ideas]
I love the handwriting style comments, but...
Please, please, I need the standard text commenting feature to be available as well (like in any other PDF viewer).
I.e. I would like to be able to select a text from PDF and annotate it with a (collapsible/pop-up?) text comment that I type in as text.
Standard uses of it would be:
1. formal document reviews - to provide review comments to the author (who isn't necessarily capable of decrypting my handwriting). Currently the highlighter + text tool kind-of works, but I wish I could:
  a) select the text to highlight (or underline or strike through)
  b) attach a (possibly very long) comment to the above highlighted text
  c) clearly have a list of comments available
With my current solution I don't like the always visible floating text mixed with the original text and I can't easily see and jump to the comments. It should be possible to display the list of comments separately, e.g. in a separate panel. Ideally, I would also be able to export such comments as text (including the original highlighted text) and see when and by whom these were written.
2. text notes for myself when following lecture, presentation, etc. (I am much faster typing text than handwriting it)

That would really make me super-happy and I would use xournal for everything then.

**Comment by rolandlo** [comment]
You should probably make your ideas heard in PR #2660.

**Comment by indigolark** [comment]
Annotating pdfs with handwriting works very well for me, but text comment boxes would be very useful too. Such a feature would be very handy for teachers who get a lot of digital assignments nowadays . I use xournalpp on a Wacom tablet for text marking and pdf annotation, and have to write the same comments over and over again. Having the ability to create a collection of reusable comments would make my correction work easier. Up to now I haven't found an application that can handle both handwriting and reusable comments.

**Comment by turbotimon** [comment]
Any news on that? I love Xournal, but also miss this feature. For context, i think we are talking about a text annotation popup similar to this, right?:

![image](https://github.com/xournalpp/xournalpp/assets/57718207/5a56c357-0b5d-475e-a991-a77b2d26912f)

**Comment by vdunias** [comment]
Hello,

This would be a very helpful toll to have this improvement.

I am a math teacher and use Xournal to annotate student copies.
Some errors are recurring and force me to repeat the same thing on several copies.
To save correction time, I would love if Xournal got some new feature allowing me to quickly insert a text comment or an hyperlink on the student's copy.
It could be a tool call "Predefined annotation" allowing :
-1st step: The first click opens a dialog box containing the list of comments
-2nd step (optional): Enter one or more keywords in a search bar to filter the elements of the displayed list.
-3rd step: Select the desired comment.
The dialog box then closes, and the (hyper)link is placed at the location of the first click.

I am talking about a hyperlink but we can imagine other types of annotations: text, an image, a latex formula, etc.
